cms/djangoapps/contentstore/rest_api/v0/serializers/tabs.py

```diff
@@ -0,0 +1,109 @@
+    )
+    tab_id = serializers.CharField(
+        read_only=True,
+        help_text=_("Name of the tab displayed to users. Overrides title."),
+    )
+    settings = serializers.DictField(
+        read_only=True,
+        help_text=_("Additional settings specific to the tab"),
+    )
+
+    def to_representation(self, instance: CourseTab):
+        tab_data = {
+            "type": instance.type,
+            "title": instance.title,
+            "is_hideable": instance.is_hideable,
+            "is_hidden": instance.is_hidden,
+            "is_movable": instance.is_movable,
+            "course_staff_only": instance.course_staff_only,
+            "name": instance.name,
+            "tab_id": instance.tab_id,
+        }
+        tab_settings = {
+            key: value for key, value in instance.tab_dict.items() if key not in tab_data and key != "link_func"
+        }
+        tab_data["settings"] = tab_settings
+        return tab_data
+
+
+class TabIDLocatorSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """Serializer for tab locations, used to identify a tab in a course."""
+
+    tab_id = serializers.CharField(required=False, help_text=_("ID of tab to update"))
+    tab_locator = UsageKeyField(required=False, help_text=_("Location (Usage Key) of tab to update"))
+
+    def validate(self, attrs):
+        if "tab_id" not in attrs and "tab_locator" not in attrs:
+            raise serializers.ValidationError("At least one of tab_id and tab_locator is required")
+        return super().validate(attrs)
+
+
+class CourseTabUpdateSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer to update course tabs.
+    """
+
+    tab_id_locator = TabIDLocatorSerializer(
+        required=False,
+        help_text=_("A tab location expressed via either a `tab_id` or `tab_location`"),
+    )
+    is_hidden = serializers.BooleanField(
+        required=False,
+        help_text=_("True to hide the tab, and False to show it."),
+    )
+    tabs = serializers.ListField(
+        required=False,
+        child=TabIDLocatorSerializer(help_text=_("A tab location expressed via either a `tab_id` or `tab_location`")),
+        help_text=_("A list of tab locators in the order in which they should appear in the UI."),
+        min_length=2,
+    )
+
+    def validate(self, attrs):
+        """Validate the the correct combination of fields has been provided via the API."""
+        tab_id_locator = "tab_id_locator" in attrs
+        is_hidden = "is_hidden" in attrs
+        tabs = "tabs" in attrs
+        if (
+            # Either supply a `tabs` field or the `tab_id_locator` and `is_hidden` fields
+            (tabs and (is_hidden or tab_id_locator))
+            or (not tabs and not (is_hidden and tab_id_locator))
+        ):
+            raise serializers.ValidationError(
+                _(
+                    "Need to supply either an ordered list of tab locators via `tabs` or "
```

Comment:
Can we move that error message to a const/enum so that it's easier to change/update in the future? Currrently, if this changes we'd need to update it in the test too as its hardcoded there.

Reply:
I think that would run counter to testing. If the actual error message doesn't matter we shouldn't test it. If it does matter then the message changing should be an error. In this case, since the error isn't user-facing, I'll assert the presence of an error without asserting about the specifics.